tests/src/test/scala/tests/TypoSuite.scala

```diff
@@ -194,21 +194,27 @@ class TypoSuite extends FunSuite {
 
   // extension methods to abstract over lsp4s service creation.
   implicit class XtensionServices(services: Services) {
-    def forwardRequest[A, B](endpoint: Endpoint[A, B])(implicit client: JsonRpcClient): Services = {
-      services.requestAsync(endpoint)(a => endpoint.request(a))
+    def forwardRequest[A, B](endpoint: Endpoint[A, B])(implicit client: RpcClient): Services = {
+      //services.requestAsync(endpoint)(a => endpoint.request(a))
+
+      // FIXME  returns a Task[RpcReponse[B]] but needs a Task[B]
+      // Not fully sure how to fix this one
+
+      //services.requestAsync(endpoint)(a => endpoint.request(a))
```

Comment:
Also this is the only thing I has left which is tripping me up. `endpoint.request()` here returns a `Task[RpcResponse[B]]` but a `Task[B]` is actually needed. I'm struggling with getting that to that point. Any pointers about what I'm missing here would be helpful.

Reply:
Yeah, this is a bit unintuitive if you don't know about the underlying jsonrpc changes. I've added a code suggestion to handle this case, I'm not sure it works but it should.